src/App.js

```diff
@@ -13,78 +14,119 @@ function App() {
+  const addTodo = async (newData) => {
     try {
-      const response = await fetch(API_URL,
+      const data = await fetchApi(
+        'POST',
+        API_URL,
         {
-          method: 'POST',
-          headers: {
-            'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify({ records: [{ fields: data }] })
-        });
-
-      if (!response.ok) {
-        throw new Error(`HTTP error! Status: ${response.status}`);
-      }
-      const newData = await response.json();
-      const listItem = newData.records[0]
+          'Content-Type': 'application/json'
+        },
+        { records: [{ fields: newData }] }
+      )
+      const listItem = data.records[0]
       setTodoList(prevItems => [...prevItems, {
         title: listItem.fields.Title,
+        done: listItem.fields.Done,
         id: listItem.id,
-
       }])
     } catch (error) {
       console.error('Error updating todo:', error.message);
       return null
     }
   }
-
-  const removeTodo = async (id) => {
-    try {
-      const response = await fetch(`${API_URL}/${id}`, {
-        method: 'DELETE',
-        headers: {
-          'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
-        },
-      });
-      if (!response.ok) {
-        throw new Error(`HTTP error! Status: ${response.status}`);
+  const fadeOutElement = async (element) => {
+    element.classList.add("fade-out")
+    let opacity = 1;
+    const fadeOutInterval = setInterval(function () {
+      if (opacity > 0) {
+        opacity -= 0.1;
+        element.style.opacity = opacity;
+      } else {
+        clearInterval(fadeOutInterval);
       }
+    }, 30);
+  }
 
+  const removeTodo = async (id, e) => {
+    const fadeOut = await fadeOutElement(e.target.closest('li'))
+    try {
+      const data = await fetchApi(
+        'DELETE',
+        `${API_URL}/${id}`,
+        { 'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+      )
       setTodoList(prevTodoList => prevTodoList.filter(item => item.id !== id))
     } catch (error) {
       console.error('Error deleting todo:', error.message);
+    }
+  }
 
+  const updateTodo = async (id, updatedCheckboxValue)=>{
+    try {
+      const data = await fetchApi(
+        'PATCH',
+        `${API_URL}/${id}`,
+        {
+          'Content-Type': 'application/json'
+        },
+        {fields: {
+            Done: updatedCheckboxValue,
+          },}
+      )
+      setTodoList((prevTodoList) =>
+       prevTodoList.map((item)=> item.id === data.id ? {
+        title: data.fields.Title,
+        done: data.fields.Done,
+        id: data.id,
+      } : item))
+    } catch (error) {
+      console.error('Error updating todo:', error.message);
+      return null
     }
   }
 
+  const checkBox = (id,updatedCheckboxValue) => {
+    updateTodo(id,updatedCheckboxValue)
+  }
+
```

Comment:
Nitpick:
This checkBox callback function is only being used once. I'd recommend just using the `updateTodo` function.